src/Commands/Modules/RenameModule.php

```diff
@@ -0,0 +1,376 @@
+        $moduleManager = $moduleManagerBuilder->build();
+        $keepOld = $input->getOption('keep-old');
+
+        if (file_exists($newFolderPath)) {
+            $question = new ConfirmationQuestion(
+                'The destination folder ' . $newFolderPath . ' already exists. The folder will be removed and the module uninstalled.'
+                . PHP_EOL . 'Do you want to continue? (y for yes, n for no)?', false);
+            if (!$questionHelper->ask($input, $output, $question)) {
+                return 0;
+            }
+
+            $newModuleName = strtolower($newFullName['prefix'] . $newFullName['name']);
+            if ($moduleManager->isInstalled($newModuleName)) {
+                $newModule = Module::getInstanceByName($newModuleName);
+                if ($newModule && $newModule->uninstall()) {
+                    $io->success('The module ' . $newModuleName . ' has been uninstalled.');
+                } else {
+                    $io->error('The module ' . $newModuleName . ' couldn\'t be uninstalled.');
+
+                    return 1;
+                }
+            }
+            exec('rm -rf ' . $newFolderPath);
+        }
+
+        if (!$keepOld && $oldModule && $moduleManager->isInstalled($oldModuleName)) {
+            if ($oldModule->uninstall()) {
+                $io->success('The old module ' . $oldModuleName . ' has been uninstalled.');
+            } else {
+                $io->error('The old module ' . $oldModuleName . ' couldn\'t be uninstalled.');
+
+                return 1;
+            }
+        }
+        exec('cp -R ' . $oldFolderPath . '. ' . $newFolderPath);
+        if (!$keepOld) {
+            exec('rm -rf ' . $oldFolderPath);
+        }
+
+        $finder = new Finder();
+        $finder->exclude(['vendor', 'node_modules']);
+        foreach ($finder->in($newFolderPath) as $file) {
+            if ($file->isFile()) {
+                $fileContent = file_get_contents($file->getPathname());
+                file_put_contents($file->getPathname(), strtr($fileContent, $replace_pairs));
+            }
+            foreach ($replace_pairs as $search => $replace) {
+                if (strpos($file->getRelativePathname(), $search) !== false) {
+                    rename($file->getPathname(), $newFolderPath . strtr($file->getRelativePathname(), $replace_pairs));
+                    break;
+                }
+            }
+        }
+
+        // Composer\Factory::getHomeDir() method
+        // needs COMPOSER_HOME environment variable set
+        //putenv('COMPOSER_HOME=' . __DIR__ . '/vendor/bin/composer');
+        if (file_exists($newFolderPath . 'composer.json')) {
+            chdir($newFolderPath);
+            exec('rm -rf vendor');
+            $installCommand = new ArrayInput(['command' => 'update']);
+            $dumpautoloadCommand = new ArrayInput(['command' => 'dumpautoload', '-a']);
+            $application = new Application();
```

Comment:
The less dependencies there is the better it is (less problem).
Did you tried with `Symfony\Component\Process\Process` ? (I've used it some times and it works fine).